### DISCUSSION TITLE
How do I enable FlxMouseControl?

### DISCUSSION THREAD

**Discussion by JasperHorn** [HaxeFlixel/flixel#2867] [Q&A]
I want to use sprites that can be clicked on. FlxExtendedSprite offers this functionality, but when I try to call `extendedSprite.enableMouseClicks(true)` I get the following error:

> FlxExtendedSprite.enableMouseClicks called but FlxMouseControl plugin not activated

I was unable to find the proper way to add this plugin, but the most likely option seemed to be this:

    FlxG.plugins.add(new FlxMouseControl());

However, this gives its own error:

> flixel_FlxG.plugins.add_flixel_addons_plugin_FlxMouseControl is not a function

How do I enable this plugin?

**Comment by JasperHorn** [comment]
So eh... I think the problem resolved itself.

I implemented the clicking myself - based on code from an old project (which I never got too far with, andit's been years, so I still very much feel like I'm a Haxe(Flixel) beginner) - the FlxExtentedSprite's doesn't appear to be too special anyway . And then I realized that the code for adding the plugin was still in there, yet the error wasn't showing up anymore. So I went back to using the FlxExtendedSprite and now that just works.

I don't quite know what happened. I suspect the compilation must not entirely have gone right somehow.

**Comment by Geokureli** [comment]
Check this out: https://dev.haxeflixel.com/demos/FlxMouseEvent/

I believe FlxMouseControl is an old leftover from the Flash version of flixel (since the doc mentions photonstorm, who worked on the flash version) and less maintained than FlxMouseEvent

**Reply by JasperHorn** [reply]
Ah, interesting! That does indeed look a lot better...
